hazelcast-spring/src/main/java/com/hazelcast/spring/HazelcastConfigBeanDefinitionParser.java

```diff
@@ -335,6 +339,18 @@ private AbstractBeanDefinition handleService(Node node) {
                     }
                 }
             }
+
+            final NamedNodeMap attrs = node.getAttributes();
+            Node classNameNode = attrs.getNamedItem("class-name");
+            if (classNameNode != null) { 
+            	serviceConfigBuilder.addPropertyValue("className", getTextContent(classNameNode));
+            }
+            Node implNode = attrs.getNamedItem("implementation");
+            if (implNode != null) { 
+                serviceConfigBuilder.addPropertyReference("implementation", getTextContent(implNode));
+            }
+            Assert.isTrue(classNameSet || classNameNode != null || implNode != null, "One of 'class-name' or 'implementation' "
```

Comment:
you are checking `classNameSet` variable.. Probably it will true when configuration is valid.